raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -1629,16 +1872,29 @@ private long pollFollowerAsVoter(FollowerState state, long currentTimeMs) throws
             transitionToCandidate(currentTimeMs);
             return 0L;
         } else {
-            long backoffMs = maybeSendRequest(
-                currentTimeMs,
-                state.leaderId(),
-                this::buildFetchRequest
-            );
+            long backoffMs;
+            if (state.fetchingSnapshot().isPresent()) {
```

Comment:
Can we move this logic to `pollFollower` somehow?

Reply:
I cleaned this up by moving it to a method we can reuse. Let me know if that addresses your concern.